examples/pennylane_universal_classifier.py

```diff
@@ -0,0 +1,491 @@
+
+
+##############################################################################
+# Train and evaluate the classifier using Adam optimizer
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+num_layers = 3
+learning_rate = .5
+epochs = 10
+batch_size = 50
+
+opt = AdamOptimizer(learning_rate, beta1=0.9, beta2=0.999)
+
+# initialize random weights
+weights = np.random.random(size=(num_layers, 3))
+
+predicted_train, states_train = test(weights, X_train, y_train, state_labels)
+accuracy_train = accuracy_score(y_train, predicted_train)
+
+predicted_test, states_test = test(weights, X_test, y_test, state_labels)
+accuracy_test = accuracy_score(y_test, predicted_test)
+
+# save predictions with random weights for comparision
+initial_predictions = predicted_test
+
+loss = cost(weights, X_test, y_test, state_labels)
+num_circuit_evaluations = 0
+
+print(
+    "Epoch: {:2d} | Cost: {:3f} | Train accuracy: {:3f} | Test Accuracy: {:3f}".format(
+        0, loss, accuracy_train, accuracy_test
+    )
+)
+
+for it in range(epochs):
+    for Xbatch, ybatch in iterate_minibatches(X_train, y_train, batch_size=batch_size):
+        weights = opt.step(lambda v: cost(v, Xbatch, ybatch, state_labels), weights)
+        num_circuit_evaluations += len(Xbatch)
+
+    predicted_train, states_train = test(weights, X_train, y_train, state_labels)
+    accuracy_train = accuracy_score(y_train, predicted_train)
+    loss = cost(weights, X_train, y_train, state_labels)
+
+    predicted_test, states_test = test(weights, X_test, y_test, state_labels)
+    accuracy_test = accuracy_score(y_test, predicted_test)
+    res = [it + 1, loss, accuracy_train, accuracy_test]
+    print(
+        "Epoch: {:2d} | Loss: {:3f} | Train accuracy: {:3f} | Test accuracy: {:3f}".format(
+            *res
+        )
+    )
+
+
+##############################################################################
+# Results
+# ~~~~~~~
+print("=====================================================================")
+print("Number of training data points: {}".format(len(X_train)))
+print("Number of test data points: {}".format(len(X_test)))
+print("Calls to circuit for training: {}".format(num_circuit_evaluations))
+print(
+    "Cost: {:3f} | Train accuracy {:3f} | Test Accuracy : {:3f}".format(
+        loss, accuracy_train, accuracy_test
+    )
+)
+for i in range(len(weights)):
+    print("Trained weights for layer {}: {}".format(i, weights[i]))
+print("==========================================================================")
+
+fig, axes = plt.subplots(1, 3, figsize=(10, 3))
+plot_data(X_test, initial_predictions, fig, axes[0])
+plot_data(X_test, predicted_test, fig, axes[1])
+plot_data(X_test, y_test, fig, axes[2])
+axes[0].set_title("Predictions with random weights")
+axes[1].set_title("Predictions after training")
+axes[2].set_title("True test data")
+plt.show()
+
+
+##############################################################################
+# `pennylane.about()`
+# ~~~~~~~
```

Comment:
Maybe a bit superfluous as a heading, could just be replaced with a sentence along the lines of "This tutorial was executed with the following PennyLane version and environment"